src/Controller/PlanningJobController.php

```diff
@@ -798,17 +798,23 @@ public function checkCopy(Request $request)
     }
 
     /**
-     * @Route("/ajax/planningjob/undo", name="planningjob.undo", methods={"GET"})
+     * @Route("/ajax/planningjob/undo", name="planningjob.undo", methods={"POST"})
      */
     public function undo(Request $request)
     {
         $date = $request->get('date');
         $site = $request->get('site');
+        $CSRFToken = $request->get('CSRFToken');
 
         if (!$this->canManagePlanning($site)) {
             return $this->json('forbiden');
         }
 
+        if (!$CSRFToken or !isset($_SESSION['oups']['CSRFToken'])
+            or $CSRFToken !== $_SESSION['oups']['CSRFToken']) {
+            return $this->json('Bad CSRF token');
+        }
+
         if (!$date || !$site) {
```

Comment:
Pour ne pas utiliser cette fonction `$this->csrf_protection($request);` ?
Voir https://suivi.biblibre.com/view.php?id=33800

Reply:
correction approuvée